### PR TITLE
Implement CI/ CD pipeline

### PR DESCRIPTION
**Pull request by radiantBear**
Sets up a CI/ CD pipeline for this repository that tests all code when a PR is created/ updated and automatically deploys all code pushed to `main`. It should work as-is, but there's a few things we should agree on:

- What do we want to name the scripts for each test type?
    - Right now, unit tests are run with `npm run test:unit`
    - Similarly, integration tests are run with `npm run test:integration`
    - End-to-end tests are run with `npm run test:e2e`
- How do we want to distinguish unit and integration tests since they both use Jest? Or do we even want to?
    - Right now, I've set the unit test script to run the tests in any Jest test file prefixed with `u-` (eg `u-example.test.js`)
    - Similarly, the integration test script runs the tests in any Jest test prefixed with `i-` (eg `i-example.test.js`)

I'd argue for the second bullet that distinguishing them is valuable for making it easier to debug if a test should fail, especially since we're splitting up the work based on test types. Distinguishing file names using a method like this will also ensure that we don't encounter issues with filename collisions or inconsistently named files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,19 @@
   "description": "CS 362 Final Project",
   "scripts": {
     "build": "webpack",
-    "start": "webpack serve --open"
+    "start": "webpack serve --open",
+    "cy:open": "cypress open",
+    "cy:run": "cypress run",
+    "test:unit": "jest u-",
+    "test:integration": "jest i-",
+    "test:e2e": "start-test start http://localhost:8080 cy:run"
```
Comment (radiantBear) [inline package.json:12]: What do we think about these test script names & the filenaming restrictions they introduce?

File: package.json [hunk 2]
```diff
@@ -4,11 +4,19 @@
   "description": "CS 362 Final Project",
   "scripts": {
     "build": "webpack",
-    "start": "webpack serve --open"
+    "start": "webpack serve --open --port 3000",
+    "cy:open": "cypress open",
+    "cy:run": "cypress run",
+    "test:unit": "jest u-",
+    "test:integration": "jest i-",
+    "test:e2e": "start-test start http://localhost:3000 cy:run"
```
Comment (radiantBear) [inline package.json:12]: What do we think about these test script names & the filenaming restrictions they introduce?

Comment (Rohanthap) [reply package.json:12]: They look good to me

**Review by noahdude1203 — APPROVED** [review]

**Review by noahdude1203 — APPROVED** [review]

**Comment by radiantBear** [comment]
I updated the deployment script this afternoon to deploy to the CoE servers instead of GitHub pages. Once this PR is merged, the site should be available at [web.engr.oregonstate.edu/~bairdn/cs362_final/](https://web.engr.oregonstate.edu/~bairdn/cs362_final/).

File: package.json (hunk 2 above)
Comment (jacosajohnson) [reply package.json:12]: Sounds good to me too

**Review by jacosajohnson — APPROVED** [review]

**Pull request merged**